spec/ddtrace/contrib/rails/rack_spec.rb

```diff
@@ -341,22 +466,75 @@ def internal_server_error
     end
   end
 
-  context 'with error rendering partial template' do
-    subject { get '/error_partial' }
+  context 'with an explicitly raised 404' do
+    subject { get '/explicitly_not_found' }
+
+    it 'does not mark span as error' do
+      is_expected.to be_not_found
+
+      expect(spans).to have_at_least(1).item
+      request_span = spans[0]
+
+      expect(request_span.name).to eq('rack.request')
+      expect(request_span.span_type).to eq('web')
+      expect(request_span.resource).to eq('TestController#explicitly_not_found')
+      expect(request_span.get_tag('http.url')).to eq('/explicitly_not_found')
+      expect(request_span.get_tag('http.method')).to eq('GET')
+      expect(request_span.get_tag('http.status_code')).to eq('404')
+
+      if rails_older_than_3_2
+        # Old Rails does not have ActionDispatch::ExceptionWrapper, thus lets the error bubble up.
+        expect(request_span).to have_error
+      else
+        expect(request_span).to_not have_error
+      end
+    end
+  end
+
+  context 'with error rendering a template' do
+    subject { get '/error_template' }
 
     it 'has ActionView error tags' do
       is_expected.to be_server_error
 
-      if Gem::Version.new(Rails::VERSION::STRING) < Gem::Version.new('3.2')
-        expect(spans).to have(5).items
+      expect(spans).to have(3).items
+      request_span, controller_span, render_span = spans
+
+      expect(request_span).to have_error
+      expect(request_span).to have_error_type('ActionView::Template::Error')
+      expect(request_span).to have_error_stack
+      expect(request_span).to have_error_message
+      expect(request_span.resource).to_not eq(render_span.resource)
```

Comment:
The minispec versions of this and other tests usually had stricter assertions for the request span, e.g.:

```ruby
    assert_equal(span_request.name, 'rails.action_controller')
    assert_equal(span_request.status, 1)
    assert_equal(span_request.span_type, 'web')
    assert_equal(span_request.resource, 'TracingController#error_template')
    assert_equal(span_request.get_tag('rails.route.action'), 'error_template')
    assert_equal(span_request.get_tag('rails.route.controller'), 'TracingController')
    assert_equal(span_request.get_tag('error.type'), 'ActionView::Template::Error')
    assert_equal(span_request.get_tag('error.msg'), 'divided by 0')
```

Was the simpler version we have with RSpec intended (because maybe we already have enough coverage elsewhere...?) or is this a gap?

Reply:
I removed some of it, was it is rather redundant, specially around the same top-level `rack.request` span, when we are asserting a nested template.

I don't think there's a meaningful loss of coverage.